server/src/main/java/org/opensearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -138,57 +138,61 @@ protected void clusterManagerOperation(
                 currentSnapshots.add(snapshotInfo);
             }
 
-            final RepositoryData repositoryData;
-            if (isCurrentSnapshotsOnly(request.snapshots()) == false) {
-                repositoryData = PlainActionFuture.get(fut -> repositoriesService.getRepositoryData(repository, fut));
-                for (SnapshotId snapshotId : repositoryData.getSnapshotIds()) {
-                    allSnapshotIds.put(snapshotId.getName(), snapshotId);
+            final StepListener<RepositoryData> repositoryDataListener = new StepListener<>();
+            repositoryDataListener.whenComplete(repositoryData -> {
+                if (isCurrentSnapshotsOnly(request.snapshots()) == false) {
+                    repositoriesService.getRepositoryData(repository, repositoryDataListener);
```

Comment:
```repositoryDataListener``` complete will never be triggered as yo are attaching the listener inside whenComplete itself.

Reply:
Ack.